File: lab1/fp_SimUDuck/DecoyDuck.cpp
#include "stdafx.h"
#include "DecoyDuck.h"

DecoyDuck::DecoyDuck()
	: Duck(FlyBehaviors::GetFlyNoWay(),
		QuackBehaviors::GetMuteQuackBehavior(),
		DanceBehaviors::GetNoDanceBehavior())
{
}

void DecoyDuck::Display() const
{
	std::cout << "I'm decoy duck" << std::endl;
}

File: lw1_fn/src/duck/DecoyDuck.php
<?php

class DecoyDuck extends Duck
{
    public function __construct()
    {
        parent::__construct(
            FlyBehavior\getFlyNoWay(),
            QuackBehavior\getMuteQuackBehavior(),
            DanceBehavior\getNoDanceBehavior());
    }

    public function display(): void
    {
        echo 'I\'m decoy duck' . PHP_EOL;
    }

    public function dance(): void
    {
    }
}